### PR TITLE
Support . workspace member

### LINKED ISSUE #1321: crates_repository fails if a workspace contains `.` as a member
This `Cargo.toml`:

```toml
[workspace]

members = [
  ".",
  "hello-lib",
]

[package]
name = "hello-rust"
version = "0.1.0"
edition = "2018"

[dependencies]
hello-lib = { path = "hello-lib" }
hex = "0.4.2"
```

currently fails to generate a repository because of this error:
```
Error: Some manifests are not being tracked. Please add the following labels to the `manifests` key: {
    "//package:Cargo.toml",
}
```

where `package` is the package containing the above `Cargo.toml`.

We should do the path normalisation to absolute paths before filtering, and normalise out any `/./` or `/../` path segments too.

### PR DESCRIPTION
**Pull request by illicitonion**
Previously, if a workspace included `.` as a member, we could
consistently error and nothing could be done about it.

Fixes #1321

### CODE REVIEW

**Comment by illicitonion** [comment]
I suggest reviewing commit-by-commit, last commit is just a re-vendoring.

File: crate_universe/src/splicing/splicer.rs [hunk 1]
```diff
@@ -87,38 +88,19 @@ impl<'a> SplicerKind<'a> {
...
-                .collect();
-
+            // UNWRAP: Safe because a Cargo.toml file must have a parent directory.
+            let root_manifest_dir = root_manifest_path.parent().unwrap();
+            let missing_manifests = Self::find_missing_manifests(
+                root_manifest,
+                root_manifest_dir,
+                &manifests
+                    .keys()
+                    .map(|p| p.normalize().expect("TODO"))
```
Comment (UebelAndre) [inline crate_universe/src/splicing/splicer.rs:98]: TODO?

File: crate_universe/src/splicing/splicer.rs [hunk 2]
```diff
@@ -1562,4 +1584,218 @@ mod test {
...
+                .normalize()
+                .unwrap(),
+            root_manifest_dir
+                .join("bar")
+                .join("Cargo.toml")
+                .normalize()
+                .unwrap(),
+        ];
+
+        {
```
Comment (UebelAndre) [inline crate_universe/src/splicing/splicer.rs:1612]: Why scope this?

File: crate_universe/src/splicing/splicer.rs [hunk 3]
```diff
@@ -148,6 +130,45 @@ impl<'a> SplicerKind<'a> {
...
+            .as_ref()
+            .unwrap()
+            .members
+            .iter()
+            .map(|member| {
+                root_manifest_dir
+                    .join(member)
+                    .join("Cargo.toml")
+                    .normalize()
+                    .expect("TODO")
```
Comment (UebelAndre) [inline crate_universe/src/splicing/splicer.rs:149]: TODO?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Looking good! Just a few nits.

**Pull request merged**